compiler/rustc_codegen_ssa/src/base.rs

```diff
@@ -905,7 +905,7 @@ pub fn provide(providers: &mut Providers) {
         };
 
         let (defids, _) = tcx.collect_and_partition_mono_items(cratenum);
-        for id in &*defids {
+        for id in defids.sorted_vector() {
```

Comment:
This is a case where a `fn any(&self, predicate: impl Fn(&T) -> bool) -> bool` directly on StableSet would allow us to skip sorting things.

Reply:
Done in https://github.com/rust-lang/rust/pull/94188/commits/95ed0262b63ca8d8760970ca04af9d5146c067af.